# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [1]:
a = 101
b = 210
c = 111
d = 402
result = (((a + b) * c)) / d
print("Результат = ", result)

Результат =  85.8731343283582


Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [2]:
ost = 11283 % 89
print("Остача від ділення числа 11283 на 89 = ", ost)

Остача від ділення числа 11283 на 89 =  69


3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [3]:
tsil_raziv = 11283 // 89
print("Результат цілочисельного ділення 11283 на 89 = ", tsil_raziv)

Результат цілочисельного ділення 11283 на 89 =  126


4. Запишіть у змінну `name` ваше імʼя.

In [4]:
name = "Daria"
name

'Daria'

5. Запишіть у змінну `surname` ваше прізвище.

In [5]:
surname = "Rudenko"
surname

'Rudenko'

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [6]:
name_surname = name + " " + surname
name_surname

'Daria Rudenko'

Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [7]:
print(name_surname.upper())

DARIA RUDENKO


Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

In [8]:
print(name_surname.lower())

daria rudenko


8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [9]:
name_surname[2]

'r'

Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [10]:
len(name_surname)

13

In [11]:
display(name_surname[0], name_surname[12])

'D'

'o'

Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [12]:
len(surname)

7

Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [13]:
capital = 'Antananarivo'
display(capital.count('A') + capital.count('a'))

4

Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [14]:
my_string = "The quick brown fox jumps over the lazy dog!!!"
print(my_string)
print(my_string.startswith("The"))
print(my_string.endswith("dog!"))

The quick brown fox jumps over the lazy dog!!!
True
False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [15]:
print(my_string.split())
print('_'.join(my_string.split()))

['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!']
The_quick_brown_fox_jumps_over_the_lazy_dog!!!


In [16]:
my_string_cut = my_string.split()
print(my_string_cut)
print('_'.join(my_string_cut))

['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!']
The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`